### PR TITLE
[Security] deprecate the $secret argument of the PersistentRememberMeHandler constructor

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | no
| New feature?  | no
| Deprecations? | yes
| Tickets       | https://github.com/symfony/symfony/pull/49103#discussion_r1086685189
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Component/Security/Http/RememberMe/PersistentRememberMeHandler.php [hunk 1]
```diff
@@ -35,8 +35,45 @@ final class PersistentRememberMeHandler extends AbstractRememberMeHandler
...
 
-    public function __construct(TokenProviderInterface $tokenProvider, #[\SensitiveParameter] string $secret, UserProviderInterface $userProvider, RequestStack $requestStack, array $options, LoggerInterface $logger = null, TokenVerifierInterface $tokenVerifier = null)
+    /**
+     * @param UserProviderInterface       $userProvider
+     * @param RequestStack                $requestStack
+     * @param array                       $options
+     * @param LoggerInterface|null        $logger
+     * @param TokenVerifierInterface|null $tokenVerifier
+     */
+    public function __construct(TokenProviderInterface $tokenProvider, $userProvider, $requestStack, $options, $logger = null, $tokenVerifier = null)
```
Comment (maxbeckers) [inline src/Symfony/Component/Security/Http/RememberMe/PersistentRememberMeHandler.php]: Shouldn't we keep the `#[\SensitiveParameter]` for the parameter `$userProvider` for the case there is injected a secret?

Comment (xabbuh) [reply src/Symfony/Component/Security/Http/RememberMe/PersistentRememberMeHandler.php]: I was wondering this myself. Would it hurt if no secret is passed an the user provider is omitted?

Comment (maxbeckers) [reply src/Symfony/Component/Security/Http/RememberMe/PersistentRememberMeHandler.php]: Yes, your solution feels a bit to be a dirty hack. I know it's not so easy to deprecate such an argument. And just sth. comming into my mind, what about using the constructor with named argument `secret`?

Comment (xabbuh) [reply src/Symfony/Component/Security/Http/RememberMe/PersistentRememberMeHandler.php]: I am not sure I understand what solution you propose.

Comment (maxbeckers) [reply src/Symfony/Component/Security/Http/RememberMe/PersistentRememberMeHandler.php]: I don't propose a solution, I proposed a maybe problem, i didn't test:
```
new PersistentRememberMeHandler (
tokenProvider: $tokenProvider,
secret: $secret, // <-- would this be a problem with your PR?
userProvider: $userProvider,
requestStack: $requestStack,
options: $options
)
```

Comment (xabbuh) [reply src/Symfony/Component/Security/Http/RememberMe/PersistentRememberMeHandler.php]: Ah, you are right about this. But named arguments are not covered by our BC promise: https://symfony.com/doc/current/contributing/code/bc.html#notes

Comment (maxbeckers) [reply src/Symfony/Component/Security/Http/RememberMe/PersistentRememberMeHandler.php]: Maybe we can make it a bit more clear in the phpdoc or whereever? And I think for now we need the `#[\SensitiveParameter]` for `$userProvider`

File: src/Symfony/Component/Security/Http/RememberMe/PersistentRememberMeHandler.php [hunk 2]
```diff
@@ -35,8 +35,45 @@ final class PersistentRememberMeHandler extends AbstractRememberMeHandler
...
+
+            $userProvider = $requestStack;
+            $requestStack = $options;
+            $options = $logger;
+            $logger = $tokenVerifier;
+            $tokenVerifier = \func_num_args() > 6 ? func_get_arg(6) : null;
+        }
+
+        if (!$userProvider instanceof UserProviderInterface) {
+            throw new \TypeError(sprintf('Argument 2 passed to "%s()" must an instance of "%s", "%s" given.', __CLASS__, UserProviderInterface::class, get_debug_type($userProvider)));
```
Comment (chalasr) [inline src/Symfony/Component/Security/Http/RememberMe/PersistentRememberMeHandler.php]:
```suggestion
            throw new \TypeError(sprintf('Argument 2 passed to "%s()" must be an instance of "%s", "%s" given.', __CLASS__, UserProviderInterface::class, get_debug_type($userProvider)));
```

same for all notices below this one

**Review by chalasr — DISMISSED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**